sdk/state/integrationtests/state_test.go

```diff
@@ -810,3 +813,195 @@ func TestOpenUpdatesCoordinatedCloseCoordinateThenStartCloseByRemote(t *testing.
 	require.NoError(t, err)
 	assert.Equal(t, amount.StringFromInt64(iBalanceCheck), assetBalance(asset, initiatorEscrowResponse))
 }
+
+func TestOpenUpdatesUncoordinatedClose_recieverNotReturningSigs(t *testing.T) {
+	// Channel constants.
+	const observationPeriodTime = 20 * time.Second
+	const averageLedgerDuration = 5 * time.Second
+	const observationPeriodLedgerGap = int64(observationPeriodTime / averageLedgerDuration)
+	const formationExpiry = 5 * time.Minute
+
+	asset := state.NativeAsset
+	// native asset has no asset limit
+	rootResp, err := client.Root()
+	require.NoError(t, err)
+	distributor := keypair.Master(rootResp.NetworkPassphrase).(*keypair.Full)
+	initiator, responder := initAccounts(t, AssetParam{
+		Asset:       asset,
+		Distributor: distributor,
+	})
+	initiatorChannel, responderChannel := initChannels(t, initiator, responder)
+
+	s := initiator.EscrowSequenceNumber + 1
+	i := int64(1)
+	e := int64(0)
+	t.Log("Vars: s:", s, "i:", i, "e:", e)
+
+	// Open
+	t.Log("Open...")
+	{
```

Comment:
TIL! I had never seen curly brackets being used to separate scopes.

Any specific reason for separating the scopes here? If I understood correctly you're basically using it to isolate test cases inside this test function.

Reply:
To make it really clear where some parts start and end. It also isolates variables created in that scope so they can't be used later, which prevents me from accidentally using values for one piece in another.